packages/ui/core-components/src/lib/unsorted/viz/map/components/Legend.svelte

```diff
@@ -0,0 +1,93 @@
+<script context="module">
+	export const evidenceInclude = true;
+</script>
+
+<script>
+	import { mapContextKey } from '../constants.js';
+	import { getContext } from 'svelte';
+	import LegendIcons from './Legend Components/LegendIcons.svelte';
+	import LegendTypes from './Legend Components/LegendTypes.svelte';
+	export let legendType = undefined;
+	export let legendPosition = 'bottomLeft';
+	const positions = {
+		topLeft: 'top-[-10px] left-[-9px]',
+		topRight: 'top-[-10px] right-[-9px]',
+		bottomLeft: 'bottom-[-9px] left-[-9px]',
+		bottomRight: 'bottom-[-9px] right-[-9px]'
+	};
+	/** @type {boolean} */
+	let hideLegend = false;
+	/** @type {[any] | undefined]} */
+	let values;
+	/** @type {[string] | undefined} */
+	let colorPalette;
+	/** @type {number | undefined} */
+	let minValue;
+	/** @type {number | undefined} */
+	let maxValue;
+	/** @type {string | undefined} */
+	export let legendFmt = undefined;
+	/** @type {'left' | 'right' | 'bottom' | 'top'} */
+	let direction = 'bottom';
+
+	const map = getContext(mapContextKey);
+	let legendData;
+
+	$: legendData = map.getLegendData;
+
+	$: {
+		if ($legendData) {
+			values = $legendData.values;
+			colorPalette = $legendData.colorPalette;
+			minValue = $legendData.minValue;
+			maxValue = $legendData.maxValue;
+		}
+	}
+
+	const handleLegendClick = () => {
+		hideLegend = !hideLegend;
+	};
+
+	if (legendType === 'scalar') {
+		direction = legendPosition.includes('Left') ? 'left' : 'right';
+	} else if (legendType === 'category') {
+		direction = legendPosition.includes('bottom') ? 'bottom' : 'top';
+	}
```

Comment:
Why do we expose `direction` as a prop above if we are overriding it here?

Reply:
Sorry about that, was moving remotely to my laptop and was mid-code. I'll remove it